core/store/src/trie/mod.rs

```diff
@@ -630,11 +614,11 @@ impl KeyForStateChanges {
         &'a self,
         store: &'b Store,
     ) -> impl Iterator<
-        Item = Result<(OwningRef<Vec<u8>, [u8]>, RawStateChangesWithMetadata), std::io::Error>,
+        Item = Result<(OwningRef<Vec<u8>, [u8]>, RawStateChangesWithTrieKey), std::io::Error>,
     > + 'b {
         let prefix_len = Self::estimate_prefix_len();
         debug_assert!(self.0.len() >= prefix_len);
-        store.iter_prefix_ser::<RawStateChangesWithMetadata>(DBCol::ColStateChanges, &self.0).map(
+        store.iter_prefix_ser::<RawStateChangesWithTrieKey>(DBCol::ColStateChanges, &self.0).map(
             move |change| {
                 // Split off the irrelevant part of the key, so only the original trie_key is left.
                 let (key, state_changes) = change?;
```

Comment:
It seems that there is no need to pull the `key` anymore. Iterator can just return `RawStateChangesWithTrieKey` instead of `(OwningRef<Vec<u8>, [u8]>, RawStateChangesWithTrieKey)`. It will have a chain reaction, but it should work out just fine.